scripts/exglobal_stage_ic.sh

```diff
@@ -16,76 +28,75 @@
 }
 
 ###############################################################
-# Start staging
-
+# Start staging gefs and gfs here
 # Stage the FV3 initial conditions to ROTDIR (cold start)
+
+for MEMDIR in ${memdir[@]}; do
```

Comment:
one this is not clear to me is that at gfs we have source target and gefs we have different path for source and target. here is the example and how we do this with one loop and we have member_dir loop for gefs under each compoenents to.
if [[ "${DO_WAVE:-}" = "YES" ]]; then
  member_dir=""
  NMEM_ENS=0
  YMD=${PDY} HH=${cyc} generate_com -r COM_WAVE_RESTART
  [[ ! -d "${COM_WAVE_RESTART}" ]] && mkdir -p "${COM_WAVE_RESTART}"
  for grdID in ${waveGRD}; do  # TODO: check if this is a bash array; if so adjust
    **source="${BASE_CPLIC}/${CPL_WAVIC}/${PDY}${cyc}/wav/${grdID}/${PDY}.${cyc}0000.restart.${grdID}"
    target="${COM_WAVE_RESTART}/${PDY}.${cyc}0000.restart.${grdID}"**
    ${NCP} "${source}" "${target}"
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
  done
elif [[ "${DO_WAVE:-}" = "YES" ]]; then
  YMD=${gPDY} HH=${gcyc} generate_com -r COM_WAVE_RESTART
  [[ ! -d "${COM_WAVE_RESTART}" ]] && mkdir -p "${COM_WAVE_RESTART}"
    **for member_dir in $(seq -w 0 $((NMEM_ENS - 1))); do
    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/ice/${PDY}.${cyc}0000.restart.${waveGRD}"
    target="${COM_OCEAN_RESTART}/${PDY}.${cyc}0000.restart.${waveGRD}"
    ${NCP} "${source}" "${target}"**
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
    done
      echo "FATAL ERROR: Unsupported ocean resolution ${OCNRES}"
      rc=1
      err=$((err + rc))
fi

Reply:
For the target, it is baked into the COM variables through `MEMDIR`: https://github.com/NOAA-EMC/global-workflow/blob/e886e8aef633389d65e23cc42456827e1db2a771/parm/config/gfs/config.com#L47
https://github.com/NOAA-EMC/global-workflow/blob/e886e8aef633389d65e23cc42456827e1db2a771/parm/config/gfs/config.com#L52

For the source, you will need to add `MEMDIR` to the paths in the appropriate place. Since `MEMDIR` for the deterministic is just an empty string, it just becomes `//` in the path, which is treated the same as a single `/`.

Reply:
But we have to alteast use "for loop" on member_dir for all components or not in that case?

Reply:
All components should be within the for loop.

Reply:
I am talking about this section and member_dir loop here highlighted 
for MEMDIR in ${memdir[@]}; do

member_dir=""
NMEM_ENS=0
YMD=${PDY} HH=${cyc} generate_com -r COM_ATMOS_INPUT
[[ ! -d "${COM_ATMOS_INPUT}" ]] && mkdir -p "${COM_ATMOS_INPUT}"
  **for member_dir in $(seq -w 0 $((NMEM_ENS - 1))); do**
    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/atmos/gfs_ctrl.nc"
    target="${COM_ATMOS_INPUT}/gfs_ctrl.nc"
    ${NCP} "${source}" "${target}"
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
    for ftype in gfs_data sfc_data; do
      for tt in $(seq 1 6); do
        source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/atmos/${ftype}.tile${tt}.nc"
        target="${COM_ATMOS_INPUT}/${ftype}.tile${tt}.nc"
        rc=$?
        (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
        err=$((err + rc))
      done
    done
  done

Reply:
That one is not correct. Use the one I gave you (and it needs to be before the `generate_com`).

Reply:
okay. Thanks Walter! I am learning a lot in shell scripting coding applications.

Reply:
I think it might be best if you started over from the original, then make just the following modifications
- Make the member_dirs list at the beginning as I recommended (like you now have it)
- Take the whole rest of the code and enclose it in the `for MEMDIR in ${member_dirs[@]}; do` for-loop
- Update the source paths to include `${MEMDIR}` where appropriate (destinations already have it in the COM templates)
- Update the file names to match what is in the alt_prototype directory where necessary (just the marine components, I think)

That should be all you have to do. Once you have your member_dirs list, you do not care whether it is GFS or GEFS, or how many members there are. All that information is encoded in the list.

Reply:
Please check the modifications. 
Let me fill up the template - any suggestions?